js/integration/knockout/validation.js

```diff
@@ -1,51 +1,72 @@
-var each = require("../../core/utils/iterator").each,
-    Class = require("../../core/class"),
-    EventsMixin = require("../../core/events_mixin"),
-    ValidationEngine = require("../../ui/validation_engine"),
-    ko = require("knockout");
-
-var koDxValidator = Class.inherit({
-    ctor: function(target, option) {
-        var that = this;
-        that.target = target;
-        that.validationRules = option.validationRules;
-        that.name = option.name;
-        that.isValid = ko.observable(true);
-        that.validationError = ko.observable();
-
-        each(this.validationRules, function(_, rule) {
-            rule.validator = that;
+import { each, map } from "../../core/utils/iterator";
+import { extend } from "../../core/utils/extend";
+import Class from "../../core/class";
+import EventsMixin from "../../core/events_mixin";
+import ValidationEngine from "../../ui/validation_engine";
+import ko from "knockout";
+
+const koDxValidator = Class.inherit({
+    ctor(target, option) {
+        this.target = target;
+        this.name = option.name;
+        this.isValid = ko.observable(true);
+        this.validationError = ko.observable();
+        this.validationErrors = ko.observable();
+        this.validationStatus = ko.observable("valid");
+
```

Comment:
```suggestion
this._validationResult = {};
```
and then you can update L26:

```js
-  if(this._validationResult && this._validationResult.status === "pending") {
+  if(this._validationResult.status === "pending") {
```

Reply:
I added _validationInfo. See [L24](https://github.com/DevExpress/DevExtreme/pull/9655/files#diff-e467507876ea61a57878947822a8ca60R24).